src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -0,0 +1,337 @@
+import fetch from '../../../../utils/fetch';
+import log from '../../../../log';
+import * as streamingTransports from '../../streamingTransports';
+import * as constants from '../constants';
+
+const LOG_AREA = 'SignalrCoreTransport';
+const NOOP = () => {};
+
+/**
+ * Handles any signal-r log, and pipes it through our logging.
+ * @param message
+ */
+function handleLog(level, message) {
+    if (level < window.signalrCore.LogLevel.Error) {
+        return;
+    }
+
+    log.warn(LOG_AREA, message);
+}
+
+function buildConnection({ baseUrl, contextId, authToken, protocol }) {
+    const accessTokenFactory = () => {
+        return authToken.replace('BEARER ', '');
+    };
+    const url = `${baseUrl}/oapi/streaming?contextId=${contextId}`;
+
+    return new window.signalrCore.HubConnectionBuilder()
+        .withUrl(url, {
+            accessTokenFactory,
+        })
+        .withHubProtocol(protocol)
+        .configureLogging({
+            log: handleLog,
+        })
+        .build();
+}
+
+/*
+ * There are few inconsistencies while using different protocols
+ * It normalizes the messages received for different protocols
+ */
+function normalizeMessage(message, protocol) {
+    // JSON protocol use camel casing and message pack use pascal case
+    const referenceId = message.referenceId || message.ReferenceId;
+    let dataFormat = message.payloadFormat || message.PayloadFormat;
+    let data = message.payload || message.Payload;
+
+    // JSON format. Normalized to old streaming format for backward compatibility
+    // Message pack parses the enum and converts it to string
+    if (dataFormat === 1 || dataFormat === 'Json') {
+        dataFormat = 0;
+    }
+
+    // Protobuf format. Normalized to old streaming format for backward compatibility
+    if (dataFormat === 2 || dataFormat === 'Protobuf') {
+        dataFormat = 1;
+    }
+
+    // JSON protocol converts bytes array to base64 encoded string
+    if (protocol.name === 'json' && typeof data === 'string') {

```

Comment:
Not sure I completely understand your comment
this data format is different than the protocol
jo json protocol could get json dataformat or protobuf

Reply:
Are you saying that if the protocol is MessagePack it can still come in lower case?

I was suggesting the code be refactored to make it clear what was being received from each protocol.

Reply:
yup.. but that is fixed now